app/src/main/java/org/dslul/openboard/inputmethod/latin/setup/SetupWizardActivity.java

```diff
@@ -198,15 +207,39 @@ mStep1Bullet, findViewById(R.id.setup_step1),
 
         mActionStart = findViewById(R.id.setup_start_label);
         mActionStart.setOnClickListener(this);
+
         mActionNext = findViewById(R.id.setup_next);
         mActionNext.setOnClickListener(this);
+
         mActionFinish = findViewById(R.id.setup_finish);
         final Drawable finishDrawable = ContextCompat.getDrawable(this, R.drawable.ic_setup_check);
-        assert finishDrawable != null;
-        DrawableCompat.setTintList(finishDrawable, new ColorStateList(new int[][]{{android.R.attr.state_focused}, {android.R.attr.state_pressed}, {}},
-                new int[]{Color.WHITE, Color.WHITE, step1.mActivatedColor}));
+        if (finishDrawable == null) {
+            return;
+        }
+        DrawableCompat.setTintList(finishDrawable, step1.mTextColorStateList);
         mActionFinish.setCompoundDrawablesRelativeWithIntrinsicBounds(finishDrawable, null, null, null);
         mActionFinish.setOnClickListener(this);
+
+        // Set the status bar color
+        getWindow().setStatusBarColor(getResources().getColor(R.color.setup_background));
+        // Navigation bar color
+        if (!isNight && !(Build.VERSION.SDK_INT >= Build.VERSION_CODES.S)) {
+            getWindow().setNavigationBarColor(ColorUtils.setAlphaComponent(Color.GRAY, 180));
+        } else {
+            getWindow().setNavigationBarColor(getResources().getColor(R.color.setup_background));
+        }
+        // Set the icons of the status bar and the navigation bar light or dark
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            final WindowInsetsController controller = getWindow().getInsetsController();
+            if (controller == null) return;
+            if (!isNight) {
+                controller.setSystemBarsAppearance(WindowInsetsController.APPEARANCE_LIGHT_STATUS_BARS, WindowInsetsController.APPEARANCE_LIGHT_STATUS_BARS);
+                controller.setSystemBarsAppearance(WindowInsetsController.APPEARANCE_LIGHT_NAVIGATION_BARS, WindowInsetsController.APPEARANCE_LIGHT_NAVIGATION_BARS);
+            }
+        } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            final View view = getWindow().getDecorView();
+            view.setSystemUiVisibility(!isNight ? View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR : 0);
+        }
```

Comment:
I find it absurd that setting this has to be done by the app...
Anyway, does it also work with theme changes, e.g. when switching between day and night while the setup screen is showing?

Reply:
<details>
<summary><b>See screenshot</b></summary>
<br>

<img width=200 src="https://github.com/Helium314/openboard/assets/139015663/548f81b2-b748-4703-9295-7a4c78bf3270">

</details>

Reply:
La seule ligne qui est peut-être inutile est:

```java
controller.setSystemBarsAppearance(WindowInsetsController.APPEARANCE_LIGHT_NAVIGATION_BARS, WindowInsetsController.APPEARANCE_LIGHT_NAVIGATION_BARS);
```
but on my devices, the keys are much more visible with this code.

Reply:
I'm ok to have the code, I understand what it's for. I just don't understand at all why an app has to tell the system what to do so the system UI is properly visible.
It's not like that it would be hard to automatically set dark buttons when the app sets a white navigation bar color.
But whatever, it's not really like this changes anything.